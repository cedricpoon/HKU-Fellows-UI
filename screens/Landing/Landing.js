import { createMaterialTopTabNavigator } from 'react-navigation';

import { Preview, SelectCourse } from 'hkufui/screens';

export default createMaterialTopTabNavigator(
  {
    Preview: Preview,
    SelectCourse: SelectCourse
  },
  {
    initialRouteName: 'Preview',
    tabBarOptions: {
      renderIndicator: () => null,
      showLabel: false,
      style: {
        display: 'none'
      }
    }
  }
);
