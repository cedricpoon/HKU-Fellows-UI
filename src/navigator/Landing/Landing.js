import { createMaterialTopTabNavigator } from 'react-navigation';

import { Preview, SelectCourse } from 'hkufui/src/screens';

export default createMaterialTopTabNavigator(
  {
    Preview: Preview,
    SelectCourse: SelectCourse
  },
  {
    initialRouteName: 'Preview',
    drawerLabel: 'Landing',
    tabBarOptions: {
      renderIndicator: () => null,
      showLabel: false,
      style: {
        display: 'none'
      }
    }
  }
);
