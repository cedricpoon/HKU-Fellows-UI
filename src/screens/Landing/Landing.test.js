import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';

describe('Testing Landing Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Landing />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
