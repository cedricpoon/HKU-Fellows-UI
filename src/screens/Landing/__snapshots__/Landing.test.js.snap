// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Landing Screen renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "navigation": Object {
        "actions": Object {},
        "addListener": [Function],
        "dangerouslyGetParent": [Function],
        "dispatch": [Function],
        "getChildNavigation": [Function],
        "getScreenProps": [Function],
        "goBack": [Function],
        "isFocused": [Function],
        "navigate": [Function],
        "router": Object {
          "childRouters": Object {
            "Preview": null,
            "SelectCourse": null,
          },
          "getActionCreators": [Function],
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        },
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "isTransitioning": false,
          "routes": Array [
            Object {
              "key": "Preview",
              "params": undefined,
              "routeName": "Preview",
            },
            Object {
              "key": "SelectCourse",
              "params": undefined,
              "routeName": "SelectCourse",
            },
          ],
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "navigation": Object {
          "actions": Object {},
          "addListener": [Function],
          "dangerouslyGetParent": [Function],
          "dispatch": [Function],
          "getChildNavigation": [Function],
          "getScreenProps": [Function],
          "goBack": [Function],
          "isFocused": [Function],
          "navigate": [Function],
          "router": Object {
            "childRouters": Object {
              "Preview": null,
              "SelectCourse": null,
            },
            "getActionCreators": [Function],
            "getActionForPathAndParams": [Function],
            "getComponentForRouteName": [Function],
            "getComponentForState": [Function],
            "getPathAndParamsForState": [Function],
            "getScreenOptions": [Function],
            "getStateForAction": [Function],
          },
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "isTransitioning": false,
            "routes": Array [
              Object {
                "key": "Preview",
                "params": undefined,
                "routeName": "Preview",
              },
              Object {
                "key": "SelectCourse",
                "params": undefined,
                "routeName": "SelectCourse",
              },
            ],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
